nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/java/org/apache/nifi/processors/standard/TestParseCEF.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.nifi.processors.standard;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.nifi.util.MockFlowFile;
+import org.apache.nifi.util.TestRunner;
+import org.apache.nifi.util.TestRunners;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.TimeZone;
+
+
+public class TestParseCEF {
+    private SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSZ");
+
+    private final static String sample1 = "CEF:0|TestVendor|TestProduct|TestVersion|TestEventClassID|TestName|Low|" +
+            // TimeStamp, String and Long
+            "rt=Feb 09 2015 00:27:43 UTC cn3Label=Test Long cn3=9223372036854775807 " +
+            // FloatPoint and MacAddress
+            "cfp1=1.234 cfp1Label=Test FP Number smac=00:00:0c:07:ac:00 " +
+            // IPv6 and String
+            "c6a3=2001:cdba::3257:9652 c6a3Label=Test IPv6 " +
+            // IPv4
+            "destinationTranslatedAddress=123.123.123.123 " +
+            // Date without TZ
+            "deviceCustomDate1=Feb 06 2015 13:27:43 " +
+            // Integer  and IP Address (from v4)
+            "dpt=1234 agt=123.123.0.124 dlat=40.366633";
+
+    @Test
+    public void testInvalidMessage() {
```

Comment:
I can certainly think about invalid messages but generally speaking this is ParCEFone's land. In the library did I either do full validation or no validation at all (we use full validation in ParseCEF).

Happy to port the jUnits across to NIFI.

One thing worth of notice is that the positive sample above contains all the data types that ParCEFone can process (i.e. String, float, long, MacAddress, IPAddr (v4 and v6) and timestamps).